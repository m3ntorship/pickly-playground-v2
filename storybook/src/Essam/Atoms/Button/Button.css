/*************   */

/* Buttons */

/*************   */
.btn {
  @apply capitalize  flex items-center justify-center  box-border m-0 focus:outline-none;
}

.btn-primary {
  @apply bg-primary border-2 focus:border-2 focus:border-dark-btnFocus rounded-full text-white hover:bg-primary-hover m-0;
}

.btn-secondary {
  @apply bg-white border border-primary text-primary focus:border-2 focus:border-dark-btnFocus rounded-full hover:bg-grey-bg m-0;
}

.btn-text {
  @apply bg-transparent text-accent hover:text-accent-hover border-0 focus:underline p-0;
}

.btn svg,
.btn path {
  @apply fill-current;
}

.btn-large {
  @apply text-md font-bold;
}

.btn-medium {
  @apply text-base font-medium;
}

.btn-small {
  @apply text-sm font-medium;
}

.btn-large.btn-primary,
.btn-large.btn-secondary {
  @apply px-xl py-sv;
}

.btn-medium.btn-primary,
.btn-medium.btn-secondary {
  @apply px-l py-xsv text-base;
}

.btn-small.btn-primary,
.btn-small.btn-secondary {
  @apply px-m py-xxsv text-sm;
}

.btn-rounded.btn-large {
  @apply p-s;
}

.btn-rounded.btn-medium,
.btn-rounded.btn-small {
  @apply p-xs;
}

.btn span,
.btn-rounded.btn-large span,
.btn-rounded.btn-medium span,
.btn-rounded.btn-small span {
  @apply flex items-center justify-center;
}

.btn.btn-large span {
  @apply w-6 h-6;
}

.btn.btn-medium span,
.btn.btn-small span {
  @apply w-4 h-4;
}

.btn-rounded.btn-large span {
  @apply w-8 h-8;
}

.btn-rounded.btn-medium span {
  @apply w-6 h-6;
}

.btn-rounded.btn-small span {
  @apply w-4 h-4;
}

button[disabled] {
  @apply disabled:opacity-24 cursor-not-allowed;
}

.btn-primary[disabled] {
  @apply hover:bg-primary;
}

.btn-secondary[disabled] {
  @apply hover:bg-white;
}

/*************   */

/* Buttons */

/*************   */
