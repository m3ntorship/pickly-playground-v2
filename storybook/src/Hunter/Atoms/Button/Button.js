import React from "react";
import PropTypes from "prop-types";

export const button = {
  type: {
    primary: "primary",
    secondary: "secondary",
    text: "text",
  },
  size: {
    small: "small",
    medium: "medium",
    large: "large",
  },

  disabled: false,

  content: "Button",

  righticon: false,

  lefticon: false,

  onlyicon: false,
};

const Button = ({
  type = "primary",
  size = "large",
  content = "Button",
  disabled = false,
  righticon = false,
  lefticon = false,
  onlyicon = false,
}) => {
  const buttontype =
    type === button.type.primary
      ? "hover:bg-primary-hover focus:outline-none focus:ring-2 focus:ring-dark-btnFocus text-white bg-primary "
      : type === button.type.secondary
      ? "text-primary bg-white border border-primary border-solid focus:ring-2  focus:ring-dark-btnFocus focus:outline-none hover:bg-grey-bg "
      : "hover:text-accent-hover focus:underline focus:outline-none text-accent  ";

  const buttonsize =
    size === button.size.large && !onlyicon
      ? "py-3.5 px-8 font-bold text-md"
      : size === button.size.medium && !onlyicon
      ? "py-2 px-6 font-semibold  text-base "
      : "py-1.5 px-4 font-semibold text-sm";

  const buttonsizelefticon =
    size === button.size.large
      ? "py-3.5 px-8 font-bold text-md"
      : size === button.size.medium
      ? "py-2 pr-6 pl-5 font-semibold  text-base "
      : "py-1.5 pl-2.5 pr-4 font-semibold text-sm";

  const buttonsizerighticon =
    size === button.size.large
      ? "py-3.5 px-8 font-bold text-md"
      : size === button.size.medium
      ? "py-2 pl-6 pr-5 font-semibold  text-base "
      : "py-1.5 pr-2.5 pl-4 font-semibold text-sm";

  const iconsize =
    size === button.size.large
      ? "w-6 h-6"
      : button.size.medium
      ? "w-4 h-4"
      : "w-4 h-4";

  const onlyiconsize =
    size === button.size.large
      ? "p-3"
      : size === button.size.medium
      ? "p-2"
      : "p-1.5";

  return (
    <>
      {/*  */}
      <button
        className={`rounded-full flex flex-row justify-between items-center
        ${buttontype}
        ${disabled && "opacity-25 cursor-default"}
        ${!onlyicon && buttonsize}
        ${onlyicon && onlyiconsize}
        ${righticon && buttonsizerighticon}
        ${lefticon && buttonsizelefticon}
        
        `}
      >
        {lefticon && (
          <svg
            className={`${iconsize} `}
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              strokelinecap="round"
              strokelinejoin="round"
              strokewidth="2"
              d="M19 9l-7 7-7-7"
            ></path>
          </svg>
        )}

        {!onlyicon && (
          <p
            className={`
            ${righticon && "mr-2"}
            ${lefticon && "ml-2"}       
            `}
          >
            {content}
          </p>
        )}
        {onlyicon && (
          <svg
            className={`${iconsize}  `}
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              strokelinecap="round"
              strokelinejoin="round"
              strokewidth="2"
              d="M19 9l-7 7-7-7"
            ></path>
          </svg>
        )}

        {righticon && (
          <svg
            className={`${iconsize}  `}
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              strokelinecap="round"
              strokelinejoin="round"
              strokewidth="2"
              d="M19 9l-7 7-7-7"
            ></path>
          </svg>
        )}
      </button>
    </>
  );
};

Button.PropTypes = {
  type: PropTypes.oneOf([
    button.type.primary,
    button.type.secondary,
    button.type.text,
  ]),
  size: PropTypes.oneOf([
    button.size.large,
    button.size.medium,
    button.size.small,
  ]),
  disabled: PropTypes.bool,
  content: PropTypes.string,
  righticon: PropTypes.bool,
  lefticon: PropTypes.bool,
  onlyicon: PropTypes.bool,
};

export default Button;
