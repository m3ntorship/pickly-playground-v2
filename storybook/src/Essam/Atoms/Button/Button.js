import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import { ReactComponent as ArrowDown } from "../../icons/arrowDown.svg";
import "./Button.css";

export const Button = ({ btnType, btnSize, withIcon, iconType, ...props }) => {
  const btnClasses = classNames(
    "btn",
    {
      "btn-primary": btnType === "primary",
      "btn-secondary": btnType === "secondary",
      "btn-text": btnType === "text",
    },
    {
      "btn-large": btnSize === "large",
      "btn-medium": btnSize === "medium",
      "btn-small": btnSize === "small",
    },
    { "btn-rounded": withIcon && iconType === "icon-only" }
  );

  return (
    <button className={btnClasses} type="button" {...props}>
      {withIcon && iconType === "icon-left" ? (
        <span className="-ml-xs mr-xs">
          <ArrowDown />
        </span>
      ) : null}

      {(btnType === "primary" || btnType === "secondary") &&
      withIcon &&
      iconType === "icon-only" ? (
        <span>
          <ArrowDown />
        </span>
      ) : (
        props.children
      )}
      {withIcon && iconType === "icon-right" ? (
        <span className="-mr-xs ml-xs">
          <ArrowDown />
        </span>
      ) : null}
    </button>
  );
};

Button.propTypes = {
  btnType: PropTypes.oneOf(["primary", "secondary", "text"]),
  btnSize: PropTypes.oneOf(["small", "medium", "large"]),
  disabled: PropTypes.bool,
  withIcon: PropTypes.bool,
  iconType: PropTypes.oneOf(["icon-right", "icon-left", "icon-only"]),
};

Button.defaultProps = {
  btnType: "primary",
  btnSize: "large",
  disabled: false,
  withIcon: false,
  iconType: "",
};
