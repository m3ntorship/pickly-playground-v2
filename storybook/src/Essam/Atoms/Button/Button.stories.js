import React from "react";

import { Button } from "./Button";

export default {
  title: "Essam/Atoms/Buttons",
  component: Button,
};

const Template = (args) => <Button {...args}></Button>;

export const DefaultButton = Template.bind({});
DefaultButton.args = {
  children: "Button Text",
  btnType: "primary",
  btnSize: "large",
  withIcon: true,
  iconType: "icon-right",
  disabled: false,
};
