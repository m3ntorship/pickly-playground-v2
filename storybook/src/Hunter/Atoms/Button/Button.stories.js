import React from "react";
import Button from "./Button";

export default {
  title: "Hunter/Atom/Button",
  component: Button,
  argTypes: {
    size: {
      control: {
        type: "radio",
        options: ["large", "medium", "small"],
      },
    },
    type: {
      control: {
        type: "radio",
        options: ["primary", "secondary", "text"],
      },
    },
  },
};

const Template = (args) => <Button {...args} />;

export const button = Template.bind({});
button.args = {
  type: "primary",
  size: "large",
  content: "button",
  disabled: false,
  righticon: false,
  lefticon: false,
  onlyicon: false,
};
