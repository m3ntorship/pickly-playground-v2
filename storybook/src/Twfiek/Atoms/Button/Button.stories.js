import React from "react";
import Button from "./Button";

const defaultExport = {
  title: "Twfiek/Atoms/Button",
  component: Button,
};

export default defaultExport;

const Template = (args) => <Button {...args} />;

export const Default = Template.bind({});
Default.args = {
  size: "lg",
  variant: "primary",
  disabled: false,
  leftIcon: false,
  rightIcon: false,
  children: "Button Text",
};
