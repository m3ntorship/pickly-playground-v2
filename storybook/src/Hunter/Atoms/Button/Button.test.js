import React from "react";
import Button from "./Button";
import userEvent from "@testing-library/user-event";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

describe("Button component", () => {
  it("should render (primary-large) button when we apply nothing", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (large) button when we apply (large)", () => {
    const tree = renderer.create(<Button size="large" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (meduim) button when we apply (meduim)", () => {
    const tree = renderer.create(<Button size="meduim" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (small) button when we apply (small)", () => {
    const tree = renderer.create(<Button size="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (primary) button when we apply primary", () => {
    const tree = renderer.create(<Button type="primary" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (secondery) button when we apply (secondery)", () => {
    const tree = renderer.create(<Button type="secondery" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (text) button when we apply text", () => {
    const tree = renderer.create(<Button type="text" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (disapled) button when we apply disapled", () => {
    const tree = renderer.create(<Button disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (righticon) button when we apply righticon", () => {
    const tree = renderer.create(<Button righticon />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (lefticon) button when we apply lefticon", () => {
    const tree = renderer.create(<Button lefticon />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (onlyicon)  button when we apply nothing", () => {
    const tree = renderer.create(<Button onlyicon />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("test button component", () => {
  it("should be disabled (not active) ", () => {
    const { getByRole } = render(<Button disabled />);
    const button = getByRole("button");
    expect(button).toBeDisabled();
  });

  it("should run a function when clicked", () => {
    const mockFunction = jest.fn();
    const { getByText } = render(<Button onClick={mockFunction} />);
    const button = getByText("button");
    userEvent.click(button);
    expect(mockFunction).toHaveBeenCalled();
  });
});
