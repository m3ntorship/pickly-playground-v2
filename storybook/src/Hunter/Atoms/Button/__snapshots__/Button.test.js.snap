// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render (disapled) button when we apply disapled 1`] = `
<button
  className="rounded-full flex flex-row justify-between items-center
        hover:bg-primary-hover focus:outline-none focus:ring-2 focus:ring-dark-btnFocus text-white bg-primary 
        opacity-25 cursor-default
        py-3.5 px-8 font-bold text-md
        false
        false
        false
        
        "
>
  <p
    className="
            false
            false       
            "
  >
    Button
  </p>
</button>
`;

exports[`Button component should render (large) button when we apply (large) 1`] = `
<button
  className="rounded-full flex flex-row justify-between items-center
        hover:bg-primary-hover focus:outline-none focus:ring-2 focus:ring-dark-btnFocus text-white bg-primary 
        false
        py-3.5 px-8 font-bold text-md
        false
        false
        false
        
        "
>
  <p
    className="
            false
            false       
            "
  >
    Button
  </p>
</button>
`;

exports[`Button component should render (lefticon) button when we apply lefticon 1`] = `
<button
  className="rounded-full flex flex-row justify-between items-center
        hover:bg-primary-hover focus:outline-none focus:ring-2 focus:ring-dark-btnFocus text-white bg-primary 
        false
        py-3.5 px-8 font-bold text-md
        false
        false
        py-3.5 px-8 font-bold text-md
        
        "
>
  <svg
    className="w-6 h-6 "
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19 9l-7 7-7-7"
      strokelinecap="round"
      strokelinejoin="round"
      strokewidth="2"
    />
  </svg>
  <p
    className="
            false
            ml-2       
            "
  >
    Button
  </p>
</button>
`;

exports[`Button component should render (meduim) button when we apply (meduim) 1`] = `
<button
  className="rounded-full flex flex-row justify-between items-center
        hover:bg-primary-hover focus:outline-none focus:ring-2 focus:ring-dark-btnFocus text-white bg-primary 
        false
        py-1.5 px-4 font-semibold text-sm
        false
        false
        false
        
        "
>
  <p
    className="
            false
            false       
            "
  >
    Button
  </p>
</button>
`;

exports[`Button component should render (onlyicon)  button when we apply nothing 1`] = `
<button
  className="rounded-full flex flex-row justify-between items-center
        hover:bg-primary-hover focus:outline-none focus:ring-2 focus:ring-dark-btnFocus text-white bg-primary 
        false
        false
        p-3
        false
        false
        
        "
>
  <svg
    className="w-6 h-6  "
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19 9l-7 7-7-7"
      strokelinecap="round"
      strokelinejoin="round"
      strokewidth="2"
    />
  </svg>
</button>
`;

exports[`Button component should render (primary) button when we apply primary 1`] = `
<button
  className="rounded-full flex flex-row justify-between items-center
        hover:bg-primary-hover focus:outline-none focus:ring-2 focus:ring-dark-btnFocus text-white bg-primary 
        false
        py-3.5 px-8 font-bold text-md
        false
        false
        false
        
        "
>
  <p
    className="
            false
            false       
            "
  >
    Button
  </p>
</button>
`;

exports[`Button component should render (primary-large) button when we apply nothing 1`] = `
<button
  className="rounded-full flex flex-row justify-between items-center
        hover:bg-primary-hover focus:outline-none focus:ring-2 focus:ring-dark-btnFocus text-white bg-primary 
        false
        py-3.5 px-8 font-bold text-md
        false
        false
        false
        
        "
>
  <p
    className="
            false
            false       
            "
  >
    Button
  </p>
</button>
`;

exports[`Button component should render (righticon) button when we apply righticon 1`] = `
<button
  className="rounded-full flex flex-row justify-between items-center
        hover:bg-primary-hover focus:outline-none focus:ring-2 focus:ring-dark-btnFocus text-white bg-primary 
        false
        py-3.5 px-8 font-bold text-md
        false
        py-3.5 px-8 font-bold text-md
        false
        
        "
>
  <p
    className="
            mr-2
            false       
            "
  >
    Button
  </p>
  <svg
    className="w-6 h-6  "
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19 9l-7 7-7-7"
      strokelinecap="round"
      strokelinejoin="round"
      strokewidth="2"
    />
  </svg>
</button>
`;

exports[`Button component should render (secondery) button when we apply (secondery) 1`] = `
<button
  className="rounded-full flex flex-row justify-between items-center
        hover:text-accent-hover focus:underline focus:outline-none text-accent  
        false
        py-3.5 px-8 font-bold text-md
        false
        false
        false
        
        "
>
  <p
    className="
            false
            false       
            "
  >
    Button
  </p>
</button>
`;

exports[`Button component should render (small) button when we apply (small) 1`] = `
<button
  className="rounded-full flex flex-row justify-between items-center
        hover:bg-primary-hover focus:outline-none focus:ring-2 focus:ring-dark-btnFocus text-white bg-primary 
        false
        py-1.5 px-4 font-semibold text-sm
        false
        false
        false
        
        "
>
  <p
    className="
            false
            false       
            "
  >
    Button
  </p>
</button>
`;

exports[`Button component should render (text) button when we apply text 1`] = `
<button
  className="rounded-full flex flex-row justify-between items-center
        hover:text-accent-hover focus:underline focus:outline-none text-accent  
        false
        py-3.5 px-8 font-bold text-md
        false
        false
        false
        
        "
>
  <p
    className="
            false
            false       
            "
  >
    Button
  </p>
</button>
`;
